x-pack/plugins/observability/server/domain/services/date_range.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { assertNever } from '@kbn/std';
+import moment from 'moment';
+
+import type { TimeWindow } from '../../types/models/time_window';
+import {
+  calendarAlignedTimeWindowSchema,
+  DurationUnit,
+  rollingTimeWindowSchema,
+} from '../../types/schema';
+
+export interface DateRange {
+  from: Date;
+  to: Date;
+}
+
+export const toDateRange = (timeWindow: TimeWindow, currentDate: Date = new Date()): DateRange => {
+  if (calendarAlignedTimeWindowSchema.is(timeWindow)) {
+    const unit = toMomentUnitOfTime(timeWindow.duration.unit);
+    const now = moment.utc(currentDate).startOf('minute');
```

Comment:
Any specific reason for `.startOf('minute')`?

Reply:
Good question. We rollup the SLI data every minute with the transform. Therefore going sub minutes won't change which bucket we are going to use to query the data.
Technically doing it or not changes nothing. It just brings some kind of consistence with how the data is aggregated